packages/hooks/src/createRunHook.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Returns a function which, when invoked, will execute all callbacks
+ * registered to a hook of the specified type, optionally returning the final
+ * value of the call chain.
+ *
+ * @param  {Object}   hooks          Stored hooks, keyed by hook name.
+ * @param  {?bool}    returnFirstArg Whether each hook callback is expected to
+ *                                   return its first argument.
+ *
+ * @return {Function}                Function that runs hook callbacks.
+ */
+function createRunHook( hooks, returnFirstArg ) {
+	/**
+	 * Runs all callbacks for the specified hook.
+	 *
+	 * @param  {string} hookName The name of the hook to run.
+	 * @param  {...*}   args     Arguments to pass to the hook callbacks.
+	 *
+	 * @return {*}               Return value of runner, if applicable.
+	 */
+	return function runHooks( hookName, ...args ) {
+		if ( 'string' !== typeof hookName ) {
```

Comment:
The [JavaScript coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#equality) don't recommend Yoda conditions.  In fact, the documentation and examples are mostly written in the opposite style.

Requiring `===` and `!==` removes the need for yoda conditions anyway, so I would like it if we could consider that requirement as entirely outdated anyway.